### ISSUE TITLE
XMLStreamException: Unbound namespace URI ''

### ISSUE THREAD

**Issue by archiecobbs** [FasterXML/woodstox#17]
Somehow this project got onto my classpath, replacing the default openJDK implemenation, and code that was previously working fine started failing with this exception:

```
Caused by: javax.xml.stream.XMLStreamException: Unbound namespace URI ''
	at com.ctc.wstx.sw.BaseStreamWriter.throwOutputError(BaseStreamWriter.java:1473)
	at com.ctc.wstx.sw.SimpleNsStreamWriter.writeAttribute(SimpleNsStreamWriter.java:84)
```
The `namespaceURI` being passed is empty string (i.e., default namespace).

Is the exception behavior correct?

If the behavior in this case is unspecified, it would be nice to make it compatible with the default implemenation.

Also, the behavior is inconsistent with the behavior of the version of `writeAttribute()` that does not take a `namespaceURI` parameter.

In other words:
```
output.writeAttribute("", "name", "value");
```
throws an exception while
```
output.writeAttribute("name", "value");
```
does not... but shouldn't these be equivalent?

**Comment by cowtowncoder** [comment]
@archiecobbs I think a full reproducible test case would be necessary, but from what you include I would agree with you: for attributes (but not elements) "empty" namespace should always match with no prefix case (that is, attributes never use the default namespace regardless of its binding).

Also: which version of Woodstox is this with?

**Comment by archiecobbs** [comment]
From Maven output:
```
[DEBUG]       org.codehaus.woodstox:wstx-asl:jar:3.2.6:compile
[DEBUG]          stax:stax-api:jar:1.0.1:compile
```

**Comment by archiecobbs** [comment]
Test case here:

https://gist.github.com/archiecobbs/35f7bf33d788b610debe5de882cd5649

Here's what I get:
```
$ ./run.sh 
+ CP=.
+ CP=.:stax-api-1.0.1.jar
+ CP=.:stax-api-1.0.1.jar:wstx-asl-3.2.6.jar
+ CP=.:stax-api-1.0.1.jar:wstx-asl-3.2.6.jar:wstx-asl-3.2.6.jar
+ javac -classpath .:stax-api-1.0.1.jar:wstx-asl-3.2.6.jar:wstx-asl-3.2.6.jar Issue17.java
+ java -classpath .:stax-api-1.0.1.jar:wstx-asl-3.2.6.jar:wstx-asl-3.2.6.jar Issue17
Exception in thread "main" javax.xml.stream.XMLStreamException: Unbound namespace URI ''
	at com.ctc.wstx.sw.BaseStreamWriter.throwOutputError(BaseStreamWriter.java:1473)
	at com.ctc.wstx.sw.SimpleNsStreamWriter.writeAttribute(SimpleNsStreamWriter.java:84)
	at Issue17.main(Issue17.java:21)
```

**Comment by cowtowncoder** [comment]
@archiecobbs Thank you for the test case. Unfortunately 3.2 is old enough that we do not support it any more. But if this is reproducible with 4.4 or 5.0, it could be fixed (4.4 and 5.0 have very little functional difference; latter just uses and requires Java 5, instead of Java 4).

**Comment by archiecobbs** [comment]
Thanks. I cannot reproduce the problem when I replace `wstx-asl-3.2.6.jar` with `woodstox-core-asl-4.4.1.jar`.

**Comment by cowtowncoder** [comment]
@archiecobbs Ok good that this has been resolved, and thank you for verifying.
